src/pages/index.js

```diff
@@ -5,78 +5,87 @@ import Section from "../components/Section.js";
 /*                                  ELEMENTS                                  */
 /* -------------------------------------------------------------------------- */
 
+// UserInfo
+let userID;
+
 // WRAPPERS
 const profileEditModal = document.querySelector("#profile-edit-modal");
 const addCardModal = document.querySelector("#add-card-modal");
+const avatarEditModal = document.querySelector("#edit-avatar-modal");
 const profileEditForm = profileEditModal.querySelector(".modal__form");
 const addCardForm = addCardModal.querySelector(".modal__form");
-const previewImageModal = document.querySelector("#preview-image-modal");
+const avatarEditForm = avatarEditModal.querySelector(".modal__form");
 
 //BUTTONS AND OTHER DOM NODES
 const profileEditButton = document.querySelector("#profile-edit-button");
-const profileModalCloseButton = profileEditModal.querySelector(".modal__close");
+const profileEditFormButton = profileEditModal.querySelector(".modal__button");
+
 const addNewCardButton = document.querySelector(".profile__add-button");
-const addCardModalCloseButton = addCardModal.querySelector(".modal__close");
-const previewImageModalCloseButton =
-  previewImageModal.querySelector(".modal__close");
-const addCardSubmitButton = addCardModal.querySelector(".modal__button");
+const addCardFormButton = addCardForm.querySelector(".modal__button");
+
+const avatarEditButton = document.querySelector(".profile__image-overlay");
+const profileAvatar = document.querySelector("#profile-avatar");
+const avatarEditFormButton = avatarEditModal.querySelector(".modal__button");
 
 // FORM INPUT DATA
-const profileTitle = document.querySelector(".profile__title");
-const profileDescription = document.querySelector(".profile__description");
+
 const profileTitleInput = document.querySelector("#profile-title-input");
 const profileDescriptionInput = document.querySelector(
   "#profile-description-input"
 );
-const cardTitleInput = addCardForm.querySelector(".modal__form-input-title");
-const cardUrlInput = addCardForm.querySelector(".modal__form-input-URL");
 
 //CARD LIST & TEMPLATE
-const cardListEl = document.querySelector(".card__list");
-const cardTemplate =
-  document.querySelector("#card-template").content.firstElementChild;
-
+let cardList;
 const cardSelector = "#card-template";
 
 /* -------------------------------------------------------------------------- */
 /*                                  Function                                  */
 /* -------------------------------------------------------------------------- */
 
 const renderCard = (cardData) => {
-  const card = new Card(cardData, cardSelector, (name, link) =>
-    previewimagePopup.open(name, link)
+  const card = new Card(
+    cardData,
+    cardSelector,
+    userID,
+    // handleImageClick
+    (name, link) => previewimagePopup.open(name, link),
+    // handleDeleteClick
+    (cardID) => {
+      deleteCardPopup.setSubmitAction(() => {
+        api
```

Comment:
You should also change the button text here. Please add `showLoading` and `hideLoading` methods to the confirmation popup and use them

Reply:
Hi, I've revised my code to change the submit button text to "deleting.." when loading. The code seems right, but the text isn't showing up on the website and I'm still struggling to figure out why. I will go on Discord and ask the tutors for advice. thanks

Reply:
Hi! I've revised my code and the "deleting" text is showing correctly now.